tidy3d/components/geometry.py

```diff
@@ -3281,8 +3290,269 @@ def _surface_area(self, bounds: Bound) -> float:
+        return trimesh.Trimesh(**trimesh.triangles.to_kwargs(triangles))
+
+    @cached_property
+    def trimesh(self) -> trimesh.Trimesh:
+        """A ``trimesh.Trimesh`` object representing the custom surface mesh geometry."""
+        return self._triangles_to_trimesh(self.triangles)
+
+    @cached_property
+    def triangles(self) -> np.ndarray:
+        """The triangles of the surface mesh as an ``np.ndarray``."""
+        if self.mesh_dataset is None:
+            raise DataError("Can't get triangles as 'mesh_dataset' is None.")
+        return self.mesh_dataset.triangles.to_numpy()
+
+    def _surface_area(self, bounds: Bound) -> float:
+        """Returns object's surface area given bounds."""
+        # currently ignores bounds
+        return self.trimesh.area
+
+    def _volume(self, bounds: Bound) -> float:
+        """Returns object's volume given bounds."""
+        # currently ignores bounds
+        return self.trimesh.volume
+
+    @cached_property
+    def bounds(self) -> Bound:
+        """Returns bounding box min and max coordinates.
+
+        Returns
+        -------
+        Tuple[float, float, float], Tuple[float, float float]
+            Min and max bounds packaged as ``(minx, miny, minz), (maxx, maxy, maxz)``.
+        """
+        if self.mesh_dataset is None:
+            return ((-inf, -inf, -inf), (inf, inf, inf))
+        return self.trimesh.bounds
+
+    def intersections_plane(
+        self, x: float = None, y: float = None, z: float = None
+    ) -> List[Shapely]:
+        """Returns list of shapely geoemtries at plane specified by one non-None value of x,y,z.
+
+        Parameters
+        ----------
+        x : float = None
+            Position of plane in x direction, only one of x,y,z can be specified to define plane.
+        y : float = None
+            Position of plane in y direction, only one of x,y,z can be specified to define plane.
+        z : float = None
+            Position of plane in z direction, only one of x,y,z can be specified to define plane.
+
+        Returns
+        -------
+        List[shapely.geometry.base.BaseGeometry]
+            List of 2D shapes that intersect plane.
+            For more details refer to
+            `Shapely's Documentaton <https://shapely.readthedocs.io/en/stable/project.html>`_.
+        """
+
+        if self.mesh_dataset is None:
+            return []
+
+        axis, position = self.parse_xyz_kwargs(x=x, y=y, z=z)
+
+        origin = [0, 0, 0]
+        origin[axis] = position
+        normal = [0, 0, 0]
+        normal[axis] = 1
+
+        mesh = self.trimesh
+        section = mesh.section(plane_origin=origin, plane_normal=normal)
+        if section is None:
+            return []
+
+        # permute so normal is aligned with z axis
+        # and (y, z), (x, z), resp. (x, y) are aligned with (x, y)
+        perm = np.eye(4)
+        perm[3, :3] = -np.array(origin)
```

Comment:
is there any way to simplify this permutation matrix creation? it's a bit difficult to read.